pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -418,44 +418,58 @@ func compareAndAssignMemberStatuses(reqLogger logr.Logger, toolchainStatus *tool
 	return allOk
 }
 
-// replaceStatusConditions replaces Member status conditions with the new conditions
-func (r *ReconcileToolchainStatus) replaceStatusConditions(status *toolchainv1alpha1.ToolchainStatus, newConditions ...toolchainv1alpha1.Condition) error {
-	// the controller should always update at least the last updated timestamp of the status so the status should be updated regardless of whether
-	// any specific fields were updated. This way a problem with the controller can be indicated if the last updated timestamp was not updated.
+// updateStatusConditions updates ToolchainStatus status conditions with new conditions
+// the controller should always update at least the last updated timestamp of the status so the status should be updated
+// regardless of whether any specific fields were updated. This way a problem with the controller can be indicated if
+// the last updated timestamp was not updated.
+func (r *ReconcileToolchainStatus) updateStatusConditions(logger logr.Logger, status *toolchainv1alpha1.ToolchainStatus,
+	newConditions ...toolchainv1alpha1.Condition) error {
+
 	var conditionsWithTimestamps []toolchainv1alpha1.Condition
 	for _, newCondition := range newConditions {
 		condition := copyConditionWithTimestampSet(status, newCondition)
 		conditionsWithTimestamps = append(conditionsWithTimestamps, condition)
 	}
-	status.Status.Conditions = conditionsWithTimestamps
-	return r.client.Status().Update(context.TODO(), status)
+
+	status.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(status.Status.Conditions, conditionsWithTimestamps...)
```

Comment:
We can't just use this function as is because it does not update the condition if it's not changed. But we need to update the LastUpdated timestamp regardless. It's unique to ToolchainStatus conditions.
I've modified the test to prove that the current solution breaks it: https://github.com/codeready-toolchain/host-operator/pull/343/commits/4b461bacc1bb0abb019bd0482e0bfbb37b798b4d

Reply:
Fixed.